src/index.ts

```diff
@@ -154,37 +154,75 @@ export function WithFinalAuthTo<
 // @todo: implement all the other HTPipe*'s - note that each one will be slightly different based on their specifics...
 // i.e. can return bool vs not, possibly async vs sync only, mandatory vs not mandatory...
 // then the final master HTPipe will just build an object out of all the sub-HTPipe*'s
+
```

Comment:
good job addressing the gotcha...  I think since it's such a gotcha we should comment carefully
for this one:
`// left has attachData AND right has attachData AND left's return keys that exist in right's parameters are assignable to right's correspondingly`
for the next one:
```
// left has attachData and right does not
// if right has attachData, left's return keys that exist in right's parameters must be assignable to right's correspondingly
// this conditional type is necessary to disqualify left-and-right cases that fell through the first overload because of the type incompatibility so they aren't grouped in with the left-only cases
```
...and so on
- [ ] comment this - it's gonna be hard to understand when returning to it later

Reply:
come to think of it, the second and third lines might be better to put near the conditional type on the line before `Partial<Parameters<TRight['attachData']>[0]>` everywhere - that is, if prettier / tslint don't make it impossible :-/

Reply:
done
i can't put that conditional type in one type
prettier not allow that

Reply:
I don't see the second/third lines of the comment
- [ ] finish comment

Reply:
nvm